Maker-Agents-Suit-Module/OSMAgent/src/OSM.rely.function.agent.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+} from 'forta-agent';
+
+import {
+  provideFunctionCallsDetectorAgent,
+  FindingGenerator,
+} from '@nethermindeth/general-agents-module';
+import { OSM_CONTRACTS } from './utils';
+
+export const RELY_FUNCTION_SIG = 'rely(address)';
+
+const createFindingGenerator = (alertID: string): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined) =>
+    Finding.fromObject({
+      name: 'Maker OSM Contract RELY Function Agent',
+      description: 'RELY Function is called',
+      alertId: alertID,
+      severity: FindingSeverity.Medium,
+      type: FindingType.Unknown,
+      metadata: {
+        contract: metadata ? metadata.to : null,
+      },
+    });
+};
+
+export default function provideRelyFunctionAgent(
+  alertID: string,
+  contracts: string[] = OSM_CONTRACTS
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    let findings: Finding[] = [];
+
+    contracts.map(async (_contract: string) => {
+      const agentHandler = provideFunctionCallsDetectorAgent(
+        createFindingGenerator(alertID),
+        RELY_FUNCTION_SIG,
+        { to: _contract }
+      );
```

Comment:
Maybe you can construct all the handlers before creating the agent handler.
In pseudocode:
```
function provider(contracts, ...) {
    handlers = contracts.map(function_to_create_the_handlers)
    return async (txEvent) => {
        findings = []
        for h in handlers:
            findings += await h(txEvent)
        return findings
    }
}
```